### PR TITLE
Add Test for an Annotation Based Control Bus

### PR DESCRIPTION
**Pull request by garyrussell**
http://stackoverflow.com/questions/27849211/spring-integration-control-bus-configuration-via-annotations/27850695#27850695

### CODE REVIEW

**Comment by garyrussell** [comment]
Interestingly, this comment in the book:

>When using these annotations on @Bean definitions, the inputChannel must reference a declared bean; channels are not automatically declared in this case.

Does not apply; the channel __is__ created in this test case. Maybe we should take another look at why it's unreliable, or at least change the comment to say the channel "may" need to be created.

File: spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java [hunk 1]
```diff
@@ -135,6 +137,8 @@
 @DirtiesContext
 public class EnableIntegrationTests {
 
+	private static ThreadLocal<Boolean> lifecycleStopped = new ThreadLocal<>();
```
Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java]: I don't like this `ThreadLocal` approach, so I've change it to this https://github.com/artembilan/spring-integration/commit/55aa98eb93d3601af74ede0aa63f6fd9e1095cb2
Let me know and I'll merge

**Comment by artembilan** [comment]
>  why it's unreliable

See `AbstractMethodAnnotationPostProcessor#createEndpoint`. Since it is invoked from `BPP`, there is no guaranty that we are going to use that `auto-created` channel from beans, which have been populated before this `MessagingAnnotationPostProcessor`. Similar issue I have in the DSL, when I parse `IntegrationFlow` beans. As a workaround we can use `@DependsOn`.
In most cases it has a place when we use `@MessagingGateway`, define SI endpoints manually without annotation or combine XML and Annotation configurations - `setOutputChannel()`.
Most of these cases we have resolved with `late channel resolution` from producers and others.
So, yes: 
> "may" need to be created

Is good compromize to mantion in the docs.
And let's start to gather those cases, when we have that `DestinationResolutionException` and try to fix them on demand!

Anyway we can do nothing with `BPP`, because we don't have control over end-user bean definition order.

File: spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java (hunk 1 above)
Comment (garyrussell) [inline spring-integration-core/src/test/java/org/springframework/integration/configuration/EnableIntegrationTests.java]: Polish looks good, thanks; definitely better. Let's leave the docs as-is (conservative).

**Comment by artembilan** [comment]
Merged after rebase as https://github.com/spring-projects/spring-integration/commit/1ac1c734b3e68bcdb0eb4463a9c1f5fb5d00d37b

**Pull request closed without merging**